src/utils/wallet.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+
+import { crypto, HDPublicKey, HDPrivateKey, Address } from 'bitcore-lib';
+import Mnemonic from 'bitcore-mnemonic';
+import { HD_WALLET_ENTROPY } from '../constants';
+import { XPubError } from '../errors';
+import Network from '../models/network';
+import _ from 'lodash';
+
+
+const wallet = {
+  /**
+   * Verify if words passed to generate wallet are valid. In case of invalid, returns message
+   *
+   * @param {string} words Words (separated by space) to generate the HD Wallet seed
+   *
+   * @return {Object} {'valid': boolean, 'message': string, 'words': string} where 'words' is a cleaned
+   * string with the words separated by a single space
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  wordsValid(words: string): {valid: boolean, message: string, words?: string} {
+    let newWordsString = '';
+    if (_.isString(words)) {
+      // 1. Replace all non ascii chars by a single space
+      // 2. Remove one or more spaces (or line breaks) before and after the 24 words
+      // 3. Set text to lower case
+      newWordsString = words.replace(/[^A-Za-z0-9]/g, ' ').replace(/\s+/g, ' ').trim().toLowerCase();
+      const wordsArray = newWordsString.split(' ');
+      if (wordsArray.length !== 24) {
+        // Must have 24 words
+        return {'valid': false, 'message': 'Must have 24 words'};
```

Comment:
I'd raise an exception and let the user decide what to do with it. It works better for i18n as well.

Reply:
I agree. Done